### PR TITLE
Fix broken resend email validation link

### PR DESCRIPTION
**Pull request by funkenstrahlen**
This fixes issue: https://github.com/parse-community/parse-server/issues/3857 (Missing mount point in "invalid_verification_link.html")

The old approach was to use Javascript and work with regex on the url. This is a bad approach.

I therefore took another approach using templating. I discovered that `changePassword` in `PublicAPURouter` already uses (some kind of) templating. This is very basic but seems to work.

```js
  changePassword(req) {
    return new Promise((resolve, reject) => {
      const config = new Config(req.query.id);
      if (!config.publicServerURL) {
        return resolve({
          status: 404,
          text: 'Not found.'
        });
      }
      // Should we keep the file in memory or leave like that?
      fs.readFile(path.resolve(views, "choose_password"), 'utf-8', (err, data) => {
        if (err) {
          return reject(err);
        }
        data = data.replace("PARSE_SERVER_URL", `'${config.publicServerURL}'`);
        resolve({
          text: data
        })
      });
    });
  }
```

I adopted this approach for `invalidVerificationLink`.

@flovilmart This change is currently untested and just a first POC. I would love to get your comment on this.

### CODE REVIEW

**Comment by funkenstrahlen** [comment]
Also the spec needs to be fixed. I did some changes in the spec but I do not think it is correct yet.

**Comment by flovilmart** [comment]
Hey @funkenstrahlen, this approach seems to be the best one, perhaps we should cache the file in memory, instead of reading from it all the times.

**Comment by funkenstrahlen** [comment]
@flovilmart

**Comment by funkenstrahlen** [comment]
> perhaps we should cache the file in memory, instead of reading from it all the times.

**Comment by funkenstrahlen** [comment]
> perhaps we should cache the file in memory, instead of reading from it all the times.

Do you think the increased memory footprint is worth it? This page will not be accessed very frequently. But I will leave this decision to you as you know more about parse server than I do.

How can I keep it in memory? Maybe just a global variable containing the parsed page which will be initialised on first access?

**Comment by funkenstrahlen** [comment]
@flovilmart I implemented a first approach caching the template file in memory. It does not feel perfectly right to put this kind of code into the router file though. Also I am not an experienced JS developer. I am happy if you can take a look at my approach.

https://github.com/parse-community/parse-server/pull/4002/commits/7602991bc1eca017171782b721feed5305dea4f8

**Comment by flovilmart** [comment]
@funkenstrahlen can you check, it seems there are some linting issues.

**Comment by funkenstrahlen** [comment]
Will take a look at it someday next week

File: spec/EmailVerificationToken.spec.js [hunk 1]
```diff
@@ -36,8 +36,7 @@ describe("Email Verification Token Expiration: ", () => {
           request.get(sendEmailOptions.link, {
             followRedirect: false,
           }, (error, response) => {
-            expect(response.statusCode).toEqual(302);
-            expect(response.body).toEqual('Found. Redirecting to http://localhost:8378/1/apps/invalid_verification_link.html?username=testEmailVerifyTokenValidity&appId=test');
+            expect(response.body).toEqual('Found. Redirecting to http://localhost:8378/1/apps/invalid_verification_link');
```
Comment (flovilmart) [inline spec/EmailVerificationToken.spec.js]: why are those changed? That seems to be a breaking change

**Comment by funkenstrahlen** [comment]
I fixed the syntax errors, but the test cases for invalid verification link pages do still fail.

**Pull request closed without merging**